File: front/src/model/member.ts
export interface Member {
  id: number;
  email: string;
  name: string | null;
  nickname: string | null;
  profileImage: string | null;
  provider: string;
}

export interface MemberWithToken extends Member {
  token: {
    accessToken: string;
    accessTokenExp: number;
  };
}

export interface JwtResponse {
  accessToken: string;
  accessTokenExp: number;
  refreshToken: string;
  member: Member;
}

File: src/main/java/com/techit/sociallogindemo/entity/Member.java
package com.techit.sociallogindemo.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

@ToString
@Entity
@Getter
@Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PROTECTED)
public class Member {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	private String email;

	private String name;

	private String nickname;

	private String profileImage;

	private String provider;
}
